src/mcts/search.cc

```diff
@@ -1547,6 +1557,14 @@ void SearchWorker::DoBackupUpdateSingleNode(
     if (n_to_fix > 0 && !n->IsTerminal()) {
       n->AdjustForTerminal(v_delta, d_delta, m_delta, n_to_fix);
     }
+    if (n->GetN() >= solid_threshold) {
+      if (n->MakeSolid() && n == search_->root_node_) {
+        // If we make the root solid, the current_best_edge_ becomes invalid and
+        // we should repopulate it.
+        search_->current_best_edge_ =
+            search_->GetBestChildNoTemperature(search_->root_node_, 0);
```

Comment:
What if we solidified parent `p` and cleared `current_best_edge_` instead of the next iteration of the loop after `n->FinalizeScoreUpdate` (i.e., GetN() is 1 smaller)? Then the existing best move logic would be reused:
https://github.com/LeelaChessZero/lc0/blob/b4f1dbc60a2befe4e56384441ac4a91e2972354b/src/mcts/search.cc#L1575-L1580

Reply:
Solidifying p makes n an invalid pointer and it would have to be reconstructed.  I think its probably clearer the way it is.

Reply:
Also note - typically root node solidification happens extremely rarely - since tree reuse almost always means its already solid before the search even begins.